Source/WebCore/loader/soup/ResourceLoaderSoup.cpp

```diff
@@ -34,14 +34,45 @@
 #include "ResourceError.h"
 #include "SharedBuffer.h"
 #include <gio/gio.h>
+#include <optional>
+#include <wtf/SortedArrayMap.h>
 #include <wtf/glib/GRefPtr.h>
 #include <wtf/glib/GUniquePtr.h>
 #include <wtf/glib/RunLoopSourcePriority.h>
+#include <wtf/text/ASCIILiteral.h>
 #include <wtf/text/StringView.h>
 #include <wtf/text/WTFString.h>
 
 namespace WebCore {
 
+static std::optional<String> contentTypeLookUpForKnownResource(const char* filename)
+{
+    String fileName = String::fromLatin1(filename);
```

Comment:
I think we don't need to allocate a String here. We can use StringView::fromLatin1 here.

Reply:
Good idea, let me check